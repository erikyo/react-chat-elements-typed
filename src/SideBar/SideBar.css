.rce-sbar {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    padding: 10px;
    box-sizing: border-box;
    min-height: 100%;
}

.rce-sbar.light {
    background: var(--rce-mbox-light-bg);
}

.rce-sbar.dark {
    background: var(--rce-mbox-dark-bg);
    color: white;
}

.rce-sbar-item {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;
    max-width: 100%;
}

.rce-sbar-item > * {
    display: flex;
    flex-direction: column;
}

.rce-sbar-item__center {
    margin: 15px 0;
}
