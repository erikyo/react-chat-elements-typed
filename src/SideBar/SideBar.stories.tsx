import type { StoryObj } from "@storybook/react";

import SideBar from "./SideBar";

const meta = {
	component: SideBar,
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default = {
	args: {
		style: {
			width: "300px",
			height: "100vh",
		},
		top: <div>'TOP' area</div>,
		center: <div>'CENTER' area</div>,
		bottom: <div>'BOTTOM' area</div>,
	},
};

export const Dark = {
	args: {
		type: "dark",
		style: {
			width: "300px",
			height: "100vh",
		},
		top: <div>'TOP' area</div>,
		center: <div>'CENTER' area</div>,
		bottom: <div>'BOTTOM' area</div>,
	},
};

export const Light = {
	args: {
		type: "light",
		style: {
			width: "300px",
			height: "100vh",
		},
		top: <div>'TOP' area</div>,
		center: <div>'CENTER' area</div>,
		bottom: <div>'BOTTOM' area</div>,
	},
};
