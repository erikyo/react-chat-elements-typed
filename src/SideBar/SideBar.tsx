import type { FC } from "react";
import type React from "react";
import "./SideBar.css";

import classNames from "classnames";
import type { ISideBarProps } from "../types";

const SideBar: FC<ISideBarProps> = ({
	type = "dark",
	style = {},
	...props
}) => {
	return (
		<div
			className={classNames("rce-sbar", type, props.data?.className)}
			style={style}
		>
			<div className="rce-sbar-item">{props.data?.top}</div>
			<div className="rce-sbar-item rce-sbar-item__center">
				{props.data?.center}
			</div>
			<div className="rce-sbar-item">{props.data?.bottom}</div>
		</div>
	);
};

export default SideBar;
