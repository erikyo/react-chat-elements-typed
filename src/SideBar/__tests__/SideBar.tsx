import { describe, expect, it } from "vitest";
import React from "react";
import SideBar from "../SideBar";
import { render } from "@testing-library/react";

describe("Avatar component", () => {
	it("should render without issues", () => {
		const component = render(<SideBar />);
		expect(component.container).toBeTruthy();
		expect(component.container).toMatchSnapshot();
	});
});
