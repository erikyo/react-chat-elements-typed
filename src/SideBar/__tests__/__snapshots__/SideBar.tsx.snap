// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > should render without issues 1`] = `
<div>
  <div
    class="rce-sbar default"
  >
    <div
      class="rce-sbar-item"
    />
    <div
      class="rce-sbar-item rce-sbar-item__center"
    />
    <div
      class="rce-sbar-item"
    />
  </div>
</div>
`;
